### PR TITLE
Added nginx-agent-docker on debian-bullseye

### PR DESCRIPTION
**Pull request by fabriziofiorucci**
Script and Dockerfile to build a docker image with:
- NGINX Plus + NGINX Instance Manager Agent
- NGINX App Protect WAF
- API Connectivity Manager Developer Portal

### CODE REVIEW

File: nginx-agent-docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
+FROM debian:bullseye-slim
+
+ARG NMS_URL
+ARG DEVPORTAL=false
+ARG NAP_WAF=false
+
+# Initial packages setup
+RUN	apt-get -y update \
+	&& apt-get -y install -y apt-transport-https lsb-release ca-certificates wget gnupg2 wget curl \
```
Comment (lucacome) [inline nginx-agent-docker/Dockerfile:9]:
```suggestion
	&& apt-get -y install -y apt-transport-https lsb-release ca-certificates wget gnupg2 curl \
```
you have `wget` twice. Also do you need both `curl` and `wget`?

File: nginx-agent-docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# Download certificate and key from the customer portal (https://account.f5.com)
+# and copy to the build context
+RUN --mount=type=secret,id=nginx-crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	set -x \
+# Startup script
+	&& chmod +x /deployment/start.sh && touch /.dockerenv \
+# Install prerequisite packages:
+	&& apt-get -y update \
+	&& apt-get -y install apt-transport-https lsb-release ca-certificates wget gnupg2 debian-archive-keyring \
```
Comment (lucacome) [inline nginx-agent-docker/Dockerfile:27]:
```suggestion
	&& apt-get -y install debian-archive-keyring \
```
You've already installed the packages in the other `RUN`, maybe you don't need them there? I'm not sure what the agent script needs.

File: nginx-agent-docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+# Optional API Connectivity Manager DevPortal
+# https://docs.nginx.com/nginx-management-suite/admin-guides/installation/on-prem/install-guide/
+	&& if [ "$DEVPORTAL" = "true" ] ; then \
+	printf "deb https://pkgs.nginx.com/nms/debian `lsb_release -cs` nginx-plus\n" | tee /etc/apt/sources.list.d/nms.list \
+	&& apt-key adv --keyserver keyserver.ubuntu.com --recv-keys ABF5BD827BD9BF62 \
+	&& apt-get -y update \
+	&& apt-get -y install nginx-devportal nginx-devportal-ui \
+	&& echo 'DB_TYPE="sqlite"' | tee -a /etc/nginx-devportal/devportal.conf \
+	&& echo 'DB_PATH="/var/lib/nginx-devportal"' | tee -a /etc/nginx-devportal/devportal.conf; fi \
+
```
Comment (lucacome) [inline nginx-agent-docker/Dockerfile:51]: You might wanna add a cleaning step at the end to make the image smaller, removing the packages that are not needed after the build (like `lsb-release`?)  and `rm -rf /var/lib/apt/lists/*`

File: nginx-agent-docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	&& printf "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://pkgs.nginx.com/plus/debian `lsb_release -cs` nginx-plus\n" | tee /etc/apt/sources.list.d/nginx-plus.list \
+	&& wget -P /etc/apt/apt.conf.d https://cs.nginx.com/static/files/90pkgs-nginx \
+	&& apt-get -y update \
+	&& apt-get -y install nginx-plus nginx-plus-module-njs nginx-plus-module-prometheus \
+
+# Optional NGINX App Protect WAF
+	&& if [ "$NAP_WAF" = "true" ] ; then \
+	wget -qO - https://cs.nginx.com/static/keys/app-protect-security-updates.key | gpg --dearmor | tee /usr/share/keyrings/app-protect-security-updates.gpg >/dev/null \
+	&& printf "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://pkgs.nginx.com/app-protect/debian `lsb_release -cs` nginx-plus\n" | tee /etc/apt/sources.list.d/nginx-app-protect.list \
+	&& printf "deb [signed-by=/usr/share/keyrings/app-protect-security-updates.gpg] https://pkgs.nginx.com/app-protect-security-updates/debian `lsb_release -cs` nginx-plus\n" | tee -a /etc/apt/sources.list.d/nginx-app-protect.list \
```
Comment (lucacome) [inline nginx-agent-docker/Dockerfile:38]:
```suggestion
	&& printf "deb [signed-by=/usr/share/keyrings/app-protect-security-updates.gpg] https://pkgs.nginx.com/app-protect-security-updates/debian `lsb_release -cs` nginx-plus\n" >> /etc/apt/sources.list.d/nginx-app-protect.list \
```

File: nginx-agent-docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	&& wget -qO - https://cs.nginx.com/static/keys/nginx_signing.key | gpg --dearmor | tee /usr/share/keyrings/nginx-archive-keyring.gpg >/dev/null \
+	&& printf "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://pkgs.nginx.com/plus/debian `lsb_release -cs` nginx-plus\n" | tee /etc/apt/sources.list.d/nginx-plus.list \
+	&& wget -P /etc/apt/apt.conf.d https://cs.nginx.com/static/files/90pkgs-nginx \
+	&& apt-get -y update \
+	&& apt-get -y install nginx-plus nginx-plus-module-njs nginx-plus-module-prometheus \
+
+# Optional NGINX App Protect WAF
+	&& if [ "$NAP_WAF" = "true" ] ; then \
+	wget -qO - https://cs.nginx.com/static/keys/app-protect-security-updates.key | gpg --dearmor | tee /usr/share/keyrings/app-protect-security-updates.gpg >/dev/null \
+	&& printf "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://pkgs.nginx.com/app-protect/debian `lsb_release -cs` nginx-plus\n" | tee /etc/apt/sources.list.d/nginx-app-protect.list \
```
Comment (lucacome) [inline nginx-agent-docker/Dockerfile:37]:
```suggestion
	&& printf "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://pkgs.nginx.com/app-protect/debian `lsb_release -cs` nginx-plus\n" > /etc/apt/sources.list.d/nginx-app-protect.list \
```
`tee` adds to the file and stdout, but you won't see stdout during the build so you can just use `>`

**Review by lucacome — APPROVED** [review]
Left some small suggestions

File: nginx-agent-docker/Dockerfile (hunk 3 above)
Comment (fabriziofiorucci) [reply nginx-agent-docker/Dockerfile:51]: Thank you, I will apply the change in the coming days together with a couple of small optimizations

**Pull request merged**